### ISSUE TITLE
conftest test is slow

### ISSUE THREAD

**Issue by davewongillies** [instrumenta/conftest#114]
We're using conftest in our pipelines for checking policies in against terraform plan files. We've got a few policies currently (28 at last count) and all the policies have unit tests (I think all the policies have at least a couple unit tests). I'm noticing when we run conftest test that its quite slow, and seems to be loading up the unit test files too.

```
$ time echo "{}" | conftest test -

real    0m5.752s
user    0m10.922s
sys    0m0.345s
```

If I delete all the unit test `*_test.rego` files, things are significantly faster:
```
$ time echo "{}" | conftest test -

real    0m0.602s
user    0m0.913s
sys    0m0.057s

```
Running an strace it seems as though most of the time is spent in futex syscalls:

```
$ time echo "{}" | strace -c conftest test -
% time     seconds  usecs/call     calls    errors syscall
------ ----------- ----------- --------- --------- ----------------
 96.80    0.217649          73      2977       330 futex
  0.75    0.001680           5       314           sched_yield
  0.68    0.001520           2       912           madvise
  0.57    0.001279           7       188         8 newfstatat
  0.35    0.000782           7       114           rt_sigaction
  0.19    0.000420           9        48           getdents64
  0.11    0.000253           7        37           openat
  0.11    0.000238           3        72        70 epoll_ctl
  0.10    0.000236           7        34           nanosleep
  0.08    0.000185           5        37           close
  0.07    0.000160           6        25           read
  0.06    0.000126          32         4           clone
  0.05    0.000102          10        10           rt_sigprocmask
  0.03    0.000061          61         1           readlinkat
  0.02    0.000053          11         5           fcntl
  0.02    0.000045           3        14           mmap
  0.01    0.000020           1        39           epoll_pwait
  0.01    0.000014          14         1           getpid
  0.00    0.000010           1        11           fstat
  0.00    0.000010          10         1           epoll_create1
  0.00    0.000000           0         1           execve
  0.00    0.000000           0         2           sigaltstack
  0.00    0.000000           0         1           arch_prctl
  0.00    0.000000           0         1           gettid
  0.00    0.000000           0         1           sched_getaffinity
  0.00    0.000000           0         1           getrandom
------ ----------- ----------- --------- --------- ----------------
100.00    0.224843                  4851       408 total

real    0m5.738s
user    0m10.733s
sys    0m0.578s
```

And after deleting the `*_test.rego` files:
```
$ time echo "{}" | strace -c conftest test -
% time     seconds  usecs/call     calls    errors syscall
------ ----------- ----------- --------- --------- ----------------
 90.53    0.030673          69       445        46 futex
  2.02    0.000684           2       322           madvise
  1.92    0.000652           6       114           rt_sigaction
  1.16    0.000394           3       128         8 newfstatat
  0.88    0.000298           6        49           sched_yield
  0.73    0.000249           5        48           getdents64
  0.56    0.000190           3        76        74 epoll_ctl
  0.43    0.000145           4        39           openat
  0.36    0.000122           3        39           close
  0.30    0.000103          10        10           rt_sigprocmask
  0.28    0.000095          24         4           clone
  0.19    0.000064          16         4           nanosleep
  0.17    0.000057          57         1           readlinkat
  0.10    0.000034           1        29           read
  0.10    0.000033           7         5           fcntl
  0.10    0.000033          17         2           sigaltstack
  0.06    0.000022           2        12           mmap
  0.05    0.000017           1        14           epoll_pwait
  0.04    0.000013           1        13           fstat
  0.01    0.000005           5         1           gettid
  0.00    0.000000           0         1           getpid
  0.00    0.000000           0         1           execve
  0.00    0.000000           0         1           arch_prctl
  0.00    0.000000           0         1           sched_getaffinity
  0.00    0.000000           0         1           epoll_create1
  0.00    0.000000           0         1           getrandom
------ ----------- ----------- --------- --------- ----------------
100.00    0.033883                  1361       128 total

real    0m0.630s
user    0m0.882s
sys    0m0.117s
```

**Comment by jpreese** [comment]
Interesting. I can't think of reason to load in the _test files as we're taking the structured data as input and should be applying it against the actual policy.

**Comment by davewongillies** [comment]
Perhaps `*_test.rego` files shouldn't be loaded for particular commands? https://github.com/instrumenta/conftest/blob/3d405f58718da7bb4d5ae80975c91d9f5bc1e4b5/pkg/policy/compiler.go#L53

**Comment by davewongillies** [comment]
I've raised PR #118 with a fix of sorts and the results arewhat I'm expecting:

```
$  time echo "{}" | strace -c conftest test - 
% time     seconds  usecs/call     calls    errors syscall
------ ----------- ----------- --------- --------- ----------------
 93.05    0.014895          88       169         7 futex
  1.57    0.000251           2       120           rt_sigaction
  0.93    0.000149           3        46           madvise
  0.93    0.000149          30         5           clone
  0.74    0.000119           4        27           mmap
  0.59    0.000094           1       120         8 newfstatat
  0.41    0.000066           5        13           rt_sigprocmask
  0.30    0.000048          48         1           readlinkat
  0.27    0.000044           4        11           mprotect
  0.26    0.000041           7         6           sched_yield
  0.22    0.000036           4         9           read
  0.21    0.000033           1        26           getdents64
  0.15    0.000024           1        32        30 epoll_ctl
  0.14    0.000022           4         5           fcntl
  0.11    0.000017           1        17           openat
  0.08    0.000013           1        20           close
  0.01    0.000002           0         5           fstat
  0.01    0.000002           2         1           epoll_create1
  0.01    0.000002           2         1           getrandom
  0.00    0.000000           0         3           open
  0.00    0.000000           0         1           munmap
  0.00    0.000000           0         3           brk
  0.00    0.000000           0         4         4 access
  0.00    0.000000           0         1           getpid
  0.00    0.000000           0         1           execve
  0.00    0.000000           0         1           getrlimit
  0.00    0.000000           0         2           sigaltstack
  0.00    0.000000           0         1           arch_prctl
  0.00    0.000000           0         1           gettid
  0.00    0.000000           0         1           sched_getaffinity
  0.00    0.000000           0         1           set_tid_address
  0.00    0.000000           0         1           set_robust_list
  0.00    0.000000           0         3           epoll_pwait
------ ----------- ----------- --------- --------- ----------------
100.00    0.016007                   658        49 total

real	0m0.420s
user	0m0.588s
sys	0m0.046s

```

**Comment by davewongillies** [comment]
Resolved by #118
